src/adstage/pluck_api.clj

```diff
@@ -49,14 +61,72 @@
        (into {})))
 
 (defn pluck
-  "A more generic version of the datomic.api/pull. Takes an environment
-  as the first argument.
-  Second argument is a pull style query. Third argument is an entity id or
-  entity map (from a previous pull/pluck call).
-  Specific key lookup is implemented via -pluck multimethod. Default case is
-  datomic.api/pull."
+  "An extensible wrapper around `datomic.api/pull`. Takes an environment as the
+  first argument. Second argument is a pull style query. Third argument is an
+  entity id or entity map (from a previous pull/pluck call). Specific key lookup
+  is implemented via `-pluck` multimethod. Default case is `datomic.api/pull`."
   [{:keys [db] :as env} pattern eid-or-map]
   (let [init-result (if (map? eid-or-map)
                       eid-or-map
                       (d/pull db pattern eid-or-map))]
     (pick env pattern init-result)))
+
+(defmulti -pluck-many
+  "Multimethod for extending datomic.api/pull-many. The return
+  value must be a collection of tuples.
+  `[entity-id pluck-value]`"
+  (fn [k env init-results] k))
+
+(defn -pluck-many? [k]
+  (contains? (methods -pluck-many) k))
+
+(defn -pluck-many-wrapper [k env init-results]
+  (mapv (fn [[eid v]]
+          {eid {k v}})
+        (-pluck-many k env init-results)))
+
+(defn pick-many
+  "Recursively annotates the init-result with values supplied from the
+  `-pluck-many` multimethod for root level keys and `-pluck` for sub queries."
```

Comment:
Is pluck for sub queries a design choice, or current limitation that we want to improve on? Either way, we should probably document here.